contracts/modules/maker/MakerV2Loan.sol

```diff
@@ -0,0 +1,626 @@
+     * @param _debtAmount The amount of token to repay.
+     */
+    function removeDebt(
+        BaseWallet _wallet,
+        bytes32 _loanId,
+        address _debtToken,
+        uint256 _debtAmount
+    )
+        external
+        onlyWalletOwner(_wallet)
+        onlyWhenUnlocked(_wallet)
+    {
+        verifyLoanOwner(_wallet, _loanId);
+        updateStabilityFee(uint256(_loanId));
+        removeDebt(_wallet, uint256(_loanId), _debtAmount);
+        emit DebtRemoved(address(_wallet), _loanId, _debtToken, _debtAmount);
+    }
+
+    /**
+     * @dev Closes a collateralized loan by repaying all debts (plus interest) and redeeming all collateral.
+     * @param _wallet The target wallet.
+     * @param _loanId The ID of the target CDP.
+     */
+    function closeLoan(
+        BaseWallet _wallet,
+        bytes32 _loanId
+    )
+        external
+        onlyWalletOwner(_wallet)
+        onlyWhenUnlocked(_wallet)
+    {
+        verifyLoanOwner(_wallet, _loanId);
+        updateStabilityFee(uint256(_loanId));
+        closeCdp(_wallet, uint256(_loanId));
+        emit LoanClosed(address(_wallet), _loanId);
+    }
+
+    /**
+     * @dev Gets information about a loan identified by its ID.
+     * @param _loanId The ID of the target CDP.
+     * @return a status [0: no loan, 1: loan is safe, 2: loan is unsafe and can be liquidated, 3: loan exists but we are unable to provide info]
+     * and a value (in ETH) representing the value that could still be borrowed when status = 1; or the value of the collateral that should be added to
+     * avoid liquidation when status = 2.
+     */
+    function getLoan(
+        BaseWallet /* _wallet */,
+        bytes32 _loanId
+    )
+        external
+        view
+        returns (uint8 _status, uint256 _ethValue)
+    {
+        if(cdpManager.owns(uint256(_loanId)) != address(0)) {
+            return (3,0);
+        }
+        return (0,0);
+    }
+
+    /* *************************************** Other CDP methods ***************************************** */
+
+    /**
+     * @dev Lets a CDP owner transfer their CDP from their wallet to the present module so the CDP
+     * can be managed by the module.
+     * @param _wallet The target wallet.
+     * @param _loanId The ID of the target CDP.
+     */
+    function acquireLoan(
+        BaseWallet _wallet,
+        bytes32 _loanId
+    )
+        external
+        onlyWalletOwner(_wallet)
+        onlyWhenUnlocked(_wallet)
+    {
+        invokeWallet(address(_wallet), address(cdpManager), 0, abi.encodeWithSelector(CDP_GIVE, uint256(_loanId), address(this)));
+        saveLoanOwner(_wallet, _loanId);
+    }
+
+    /**
+     * @dev Lets a SCD CDP owner migrate their CDP to use the new MCD engine.
+     * Requires MKR or ETH to pay the SCD governance fee
+     * @param _wallet The target wallet.
+     * @param _cup id of the old SCD CDP to migrate
+     */
+    function migrateCdp(
```

Comment:
I think it would make the code easier to read if we use the word `Vault` everywhere for the new CDPs as it is the official name introduced by Maker, e.g. `openVault` `giveVault` etc. Would make it easier to differentiate between V1 and V2 positions (they are both CDP in the code now).

Reply:
Yes, will do the change but it's worth noting that Maker keeps referring to their vaults as "CDPs" in their codebase (e.g. see their CdpManager contract).

Reply:
Done. Though I still kept the name 'CdpManager' to be consistent with Maker's naming.